overlord/snapstate/backend/copydata.go

```diff
@@ -333,3 +333,56 @@ func (b Backend) RemoveExposedSnapHome(snapName string) error {
 
 	return firstErr
 }
+
+type XDGDirsUndoInfo struct {
+	Moved   []string
+	Created []string
+}
+
+// InitXDGDirs moves the data, config and cache directories defined in the XDG
+// Base Directory Specification to their post core22 migration location under
+// ~Snap, if they exist. If the directories don't exist, they are created.
+// The returned XDGDirsUndoInfo struct can be used to undo this operation.
+func (b Backend) InitXDGDirs(snapName string) (undoInfo *XDGDirsUndoInfo, err error) {
+	opts := &dirs.SnapDirOptions{HiddenSnapDataDir: true, MigratedToExposedHome: true}
+	users, err := allUsers(opts)
+	if err != nil {
+		return nil, err
+	}
+
+	undoInfo = &XDGDirsUndoInfo{}
+
+	srcXdgDirs := []string{".config", ".cache", ".local/share"}
+	dstXdgDirs := []string{"xdg-config", "xdg-cache", "xdg-data"}
+
+	for _, usr := range users {
+		for i, srcDir := range srcXdgDirs {
+			newHome := snap.UserExposedHomeDir(usr.HomeDir, snapName)
+			src := filepath.Join(newHome, srcDir)
+			dst := filepath.Join(newHome, dstXdgDirs[i])
```

Comment:
maybe the spec is not very clear but the xdg-* directories should live in SNAP_USER_DATA not the exposed home (SNAP_USER_HOME). Basically is something that "stays behind" but under a different set of names.

Reply:
Ah sorry, I misread it. I'll change that

Reply:
Just to document a short discussion had out of band: the end result should be that ~/Snap/name has a copy of what's in ~/.snap/data/name/current and the XDG dirs should only exist under ~/.snap/data/name/current (if any already existed before as dotted dirs then those are removed from ~/.snap/data/name/current and ~/Snap/name)